File: 프로그래머스/2019카카오겨울인턴십_크레인인형뽑기게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    tmp = -1
    stack= []
    for i in moves:
        i = i-1
        tmp = -1
        for j in range(len(board)):
            if board[j][i]!=0:
                tmp = board[j][i]
                board[j][i] = 0
                if(len(stack)!=0 and tmp == stack[-1]):
            
                    answer +=2
                    stack.pop()
                else:
                    stack.append(tmp)
                break
        
        #print(stack)
         
    return answer

File: 프로그래머스/2019카카오겨울인턴십_크레인인형뽑기게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <stack>
#include <iostream>
using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    stack <int> st;
    int size = board.size(), into;
    for(int i=0; i < moves.size(); i++){
        int turn = moves[i] -1;
        for(int j = 0; j <size; j++){
            into = board[j][turn];
            if( into != 0){
                board[j][turn] = 0;
                if(!st.empty() && into == st.top()){
                    answer+=2;
                    st.pop();
                }
                else{
                    st.push(into);
                }
                break;
            }
        }
        
    }
    return answer;
}